package app.gate;

message CommonRequest
{
    required int64 playerid = 1;
    optional string message = 2;
}
//武将升级
message HeroUpgradeRequest
{
    required int64 playerid = 1;
    required int32 hero_no = 2;
    required int32 exp = 3;
    required int32 level = 4;
}
//武将合成
message HeroComposeRequest
{
    required int64 playerid = 1;
    required int32 hero_no = 2;
    required int32 coin = 3; //消费的金币
    repeated int32 chipids = 4; //使用的碎片
}

//武将突破
message HeroBreakRequest
{
    required int64 playerid = 1;
    required int32 hero_no = 2;
    required int32 break_level = 3;
    required int32 coin = 4; //消费的金币
}

//武将献祭
message HeroSacrificeRequest
{
    repeated int32 hero_no_list = 1;
}